### PR TITLE
fix(telemetry): harden queue timestamp parsing against malformed data

### LINKED ISSUE #231: Telemetry: harden queue timestamp parsing against malformed data
## Summary
Telemetry queue timestamp parsing can panic or misbehave on malformed timestamp strings.

## Evidence
- Fixed-index string slicing in `/Users/avifen/agnix/crates/agnix-cli/src/telemetry/queue.rs:151-156`.
- Unchecked day arithmetic at `/Users/avifen/agnix/crates/agnix-cli/src/telemetry/queue.rs:178`.

## Why This Matters
Queue files are local and can be corrupted or manually edited. Parsing should fail safely, not panic.

## Expected
Malformed timestamps are handled as `None` (or safely rejected) without panic.

## Actual
Current parser assumes strict layout and valid fields, creating panic risk and incorrect conversions.

## Suggested Fix
- Replace manual slicing/parser with robust date-time parsing (`time`/`chrono`) or strict pre-validation.
- Validate ranges (month/day/hour/minute/second).
- Keep prune behavior deterministic for invalid entries.

## Acceptance Criteria
- No panic on malformed/short/non-ASCII timestamps.
- Add tests for invalid ranges and malformed UTF-8 boundaries.
- Existing valid timestamps still parse correctly.

### PR DESCRIPTION
**Pull request by avifenesh**
## Summary

- Replace panic-prone direct byte-index slicing (`ts[0..4]`) with safe `ts.get(0..4)?` in `parse_timestamp`
- Add ASCII guard to reject multi-byte UTF-8 input before byte-position access
- Add separator character validation at expected positions (dashes, T, colons)
- Add range validation: year 1970-9999, month 1-12, month-aware day bounds, hour 0-23, minute 0-59, second 0-59
- Use `checked_sub` for day arithmetic to prevent u32 underflow
- Add 6 new test functions covering non-ASCII, invalid ranges, month-specific days (Feb 30, Apr 31), wrong separators, pre-epoch years, valid boundaries, and garbage input

Fixes #231

## Test plan

- [x] `cargo test -p agnix-cli --features telemetry` - all 14 queue tests pass (8 existing + 6 new)
- [x] `cargo test -p agnix-cli` - full suite 212 tests pass
- [x] `cargo clippy -p agnix-cli --features telemetry` - no new warnings
- [x] `cargo build --release -p agnix-cli` - release build succeeds

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
     - Supports explicit null to revert to auto-detection
 
 ### Fixed
+- CLI: harden telemetry queue timestamp parsing against malformed data (#231)
+  - Replace panic-prone byte-index slicing with safe `str::get()` calls
+  - Add ASCII guard, separator validation, and range checks (year, month-aware day bounds, hour, minute, second)
+  - Use `checked_sub` for day arithmetic to prevent u32 underflow
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply CHANGELOG.md:29]: This PR only changes 2 files: `queue.rs` (the telemetry hardening fix) and `CHANGELOG.md` (the changelog entry for it). The CHANGELOG diff shows our entry being added alongside existing entries from other already-merged PRs -- those are not changes made by this PR. The diff context shows them as unchanged context lines.

File: crates/agnix-cli/src/telemetry/queue.rs [hunk 2]
```diff
@@ -144,38 +144,70 @@ impl EventQueue {
...
 
-    // Years since 1970
-    for y in 1970..year {
-        days += if is_leap_year(y) { 366 } else { 365 };
+    let year: i32 = ts.get(0..4)?.parse().ok()?;
+    let month: u32 = ts.get(5..7)?.parse().ok()?;
+    let day: u32 = ts.get(8..10)?.parse().ok()?;
+    let hour: u32 = ts.get(11..13)?.parse().ok()?;
+    let minute: u32 = ts.get(14..16)?.parse().ok()?;
+    let second: u32 = ts.get(17..19)?.parse().ok()?;
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-cli/src/telemetry/queue.rs:171]: Intentional defense-in-depth. The `.get()` calls are zero-cost (bounds check is a comparison already proven true by the length guard) and prevent future regressions if someone modifies the length check. This pattern is recommended for safety-critical parsing.

File: crates/agnix-cli/src/telemetry/queue.rs [hunk 3]
```diff
@@ -355,4 +382,77 @@ mod tests {
...
+
+    #[test]
+    fn test_parse_timestamp_rejects_garbage() {
+        assert!(parse_timestamp("not-a-timestamp-at-all").is_none());
+        assert!(parse_timestamp("xxxx-xx-xxTxx:xx:xxZ").is_none());
+        assert!(parse_timestamp("\0\0\0\0-\0\0-\0\0T\0\0:\0\0:\0\0Z").is_none());
+        assert!(parse_timestamp("9999-99-99T99:99:99Z").is_none());
+        // Very long string (valid prefix, extra trailing data is ignored)
+        let long = "2024-01-01T00:00:00Z".to_string() + &"x".repeat(10000);
+        assert!(parse_timestamp(&long).is_some());
```
[the remark this answers is not in the document]
Comment (avifenesh) [reply crates/agnix-cli/src/telemetry/queue.rs:456]: The parser intentionally only validates the first 20 bytes (the ISO 8601 timestamp). Trailing data is ignored by design -- the `chrono_timestamp()` producer always generates exactly 20 bytes, so trailing data can only come from file corruption, which is harmless here since the parsed timestamp is valid. The test documents this behavior explicitly.

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (avifenesh) [reply CHANGELOG.md:29]: This PR does not remove the VER-* entry. The CHANGELOG diff shows that entry at line 30 as an unchanged context line. Our PR only adds the new telemetry hardening entry at lines 26-29. The VER-* entry was added by a prior PR (#258) and remains unchanged.
